### PR TITLE
Callbacks as strings

### PR DESCRIPTION
**Pull request by liamhuber**
This is stacked on top of #160, but looks more complicated than it is because it also merged in #173. Once #173 is merged, just the 4 commits from 23 Jan will show here.

Basically we add the requirement that callback functions stored on `InputSignal` need to be methods of the node that owns them. In practice this has always been the case, so there is no real change, and since you can put whatever method you want on 

Further, the requirement that callbacks be `arg`-free was already stated, but now we explicitly test for it and fail hard if it's violated.

Since the callback is now guaranteed to be a method of the owning `InputSignal.node` attribute, we can safely store its name in our state (a mere string) instead of storing the actual function object. This will make storage easier later.

### CODE REVIEW

File: pyiron_workflow/channels.py [hunk 1]
```diff
@@ -777,7 +783,42 @@ def __init__(
...
+            sum(
+                1
+                for parameter in inspect.signature(func).parameters.values()
+                if (
+                    parameter.default == inspect.Parameter.empty
+                    and parameter.kind != inspect._ParameterKind.VAR_KEYWORD
+                )
+            )
+            == 0
+        )
```
Comment (samwaseda) [inline pyiron_workflow/channels.py:817]:
```suggestion
        for parameter in inspect.signature(func).parameters.values():
            if parameter.default == inspect.Parameter.empty and parameter.kind != inspect._ParameterKind.VAR_KEYWORD:
                return False
        return True
```

**Review by samwaseda — APPROVED** [review]
Looks good to me (except for just one minor comment)

File: pyiron_workflow/channels.py (hunk 1 above)
Comment (liamhuber) [reply pyiron_workflow/channels.py:817]: Nice improvement-- Early stopping and more readable. Looking at this, I think we can do it with an `all()` comprehension and further get rid of the branching return!

Comment (liamhuber) [reply pyiron_workflow/channels.py:817]: I can commit this (so you're in the history here) and do that tomorrow, but if you see this in time feel free to modify the suggestion and directly commit it. (Assuming you like the idea)

Comment (samwaseda) [reply pyiron_workflow/channels.py:817]: Ah I see. I like the idea

Comment (samwaseda) [inline pyiron_workflow/channels.py:817]:
```suggestion
        return all([
            parameter.default != inspect.Parameter.empty
            or parameter.kind == inspect.Parameter.VAR_KEYWORD
            for parameter in inspect.signature(func).parameters.values()
        ])
```

Comment (samwaseda) [reply pyiron_workflow/channels.py:817]: I changed it silently, but apparently `inspect._ParameterKind.VAR_KEYWORD` is the same as `inspect.Parameter.VAR_KEYWORD`

**Pull request merged**